scripts/aggregate.py

```diff
@@ -0,0 +1,122 @@
+        with open(file_path, "r") as f:
+            while True:
+                line = f.readline().strip()
+                if not line:
+                    break
+                result += f"|{line}|\n"
+        result += "\n"
+        return result
+    return ""
+
+def build_summary(failures: dict, failure_name: str):
+    """Builds markdown summary tables"""
+    tools = ("gcc", "g++", "gfortran")
+    result = f"|{failure_name}|{tools[0]}|{tools[1]}|{tools[2]}|Previous Hash|\n"
+    result +="|---|---|---|---|---|\n"
+    result += f"{''.join(failures[failure_name.split(' ')[0]])}"
+    result += "\n"
+    return result
+
+def failures_to_summary(failures: dict):
+    result = "# Summary\n"
+    result += get_additional_failures("failed_build.txt", "Build Failures")
+    result += get_additional_failures("failed_testsuite.txt", "Testsuite Failures")
+    result += build_summary(failures, "Resolved Failures")
+    result += build_summary(failures, "Unresolved Failures")
+    result += build_summary(failures, "New Failures")
+    result += "\n"
+    return result
+
+def assign_labels(file_name, label):
+    """Creates label for to add to the issue"""
+    file_path = os.path.join(FAILURES, f"{file_name}")
+    if os.path.exists(file_path) and os.path.getsize(file_path) > 0:
+        return label
+    return ""
+
+def failures_to_markdown(failures: dict, current_hash: str):
+    assignees = ["patrick-rivos", "kevinl-rivosinc", "ewlu"]
+    result = f"""---
+title: Testsuite Status {current_hash}
+assignees: {", ".join(assignees)}
+"""
+    labels = {"bug"}
+    labels.add(assign_labels("failed_build.txt", "build-failure"))
+    labels.add(assign_labels("failed_testsuite.txt", "testsuite-failure"))
+    if "" in labels:
+        labels.remove("")
+    result += f"labels: {', '.join(labels)}\n"
+    result += "---\n\n"
+    result += failures_to_summary(failures)
+    return result
+
+def aggregate_summary(failures: dict, file_name: str):
+    """
+    Reads file and adds the new failures to the current
+    list of failures
+    """
+    with open(file_name, "r") as f:
+        summary_flag = False
+        while True:
+            line = f.readline()
+            if line.startswith("# Summary"):
+                summary_flag = True
+                break
+        while True:
+            line = f.readline()
+            if "#" in line:
+                # exited Summary section and going to New Failures section
+                # TODO: add support for consolidating these sections
+                break
+            if "Failures" in line:
+                index = line.split("Failures")[0][1:-1]
+                continue
+            if line != "\n" and "---" not in line:
+                failures[index].add(line)
+    return failures
+    
+def parse_arguments():
+    parser = argparse.ArgumentParser(description="Testsuite Compare Options")
+    parser.add_argument(
+        "-chash",
+        "--current-hash",
+        metavar="<string>",
```

Comment:
What does metavar do?

Reply:
According to the [docs](https://docs.python.org/3/library/argparse.html#metavar) it's a different name for the argument when displayed in help